### PR TITLE
[android][tools] fix expo-go versioning for android

### PR DESCRIPTION
**Pull request by Kudo**
# Why

fix android expo go versioning for sdk 50

# How

- fix transforms
- introduce a `kotlin-metadata-stripper` tool because `jarjar` cannot rename package names in kotlin.Metadata annotations. the react-native and some 3rd party code use kotlin will read the package name from kotlin metadata. the new tool will strip the kotlin metadata from react-native aar.

# Test Plan

`et add-sdk -p android -s 50.0.0` + versioned expo go + sdk 50 NCL

# Checklist

- n/a Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/versioning/android/java-toolkits/local.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+#
+# Location of the SDK. This is only used by Gradle.
+# For customization when using a Version Control System, please read the
+# header note.
+#Thu Dec 07 20:46:06 CST 2023
+sdk.dir=/Users/kudo/01_Work/Android/sdk
```
Comment (Kudo) [inline tools/src/versioning/android/java-toolkits/local.properties:8]: remove this local file

**Review by tsapeta — APPROVED** [review]
Looks good to me, but I'm more in favor of using the unversioned code. Up to you 😉

**Review by lukmccall — APPROVED** [review]

**Review by gabrieldonadel — APPROVED** [review]
LGTM, thanks for handling this

File: android/versioned-abis/expoview-abi49_0_0/src/main/java/abi49_0_0/host/exp/exponent/core/modules/ExpoGoUpdatesModule.kt [hunk 2]
```diff
@@ -43,7 +43,7 @@ class ExpoGoUpdatesModule(experienceProperties: Map<String, Any?>) : Module() {
         constants["runtimeVersion"] = configuration.runtimeVersionRaw ?: ""
         constants["checkAutomatically"] = configuration.checkOnLaunch.toJSString()
         constants["channel"] = configuration.requestHeaders["expo-channel-name"] ?: ""
-        constants["nativeDebug"] = BuildConfig.EX_UPDATES_NATIVE_DEBUG
+        constants["nativeDebug"] = BuildConfig.DEBUG
```
Comment (wschurman) [inline android/versioned-abis/expoview-abi49_0_0/src/main/java/abi49_0_0/host/exp/exponent/core/modules/ExpoGoUpdatesModule.kt:46]: Nice catch!

```suggestion
        constants["nativeDebug"] = false
```
(for expo go having this field doesn't even make sense for the most part, I'm fine just hardcoding)

**Review by wschurman — APPROVED** [review]
expo-updates changes look good. didn't review others as closely.

**Pull request merged**